fix: empty reward bags should disappear after closing/opening reward chest

ElimarCosta opened the pull request:
# Description

1. Reward bags should disappear after closing/opening the reward chest again
2. In the official server is possible to have more than 100 reward bags, on canary, it behaves strangely if you have more than 100 reward bags and relog
3. Sometimes when you kill more than one boss at once (same second) it mixes the rewards, let's say you kill two bosses at once it will create only one reward bag with both loots

## Behaviour
### **Actual**

You need to log out/in to remove empty reward bags
There was a limit of 100 rewards after that the system behaved strangely
When 'n' bosses were killed at the same second it would create only one reward bag for 'n' bosses

### **Expected**

Empty reward bags should disappear after closing/opening
Shouldn't have limit for rewards
Each boss should have his reward bag

## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)

ElimarCosta committed:
fix

ElimarCosta committed:
fix

ElimarCosta committed:
fix

ElimarCosta committed:
fix

ElimarCosta committed:
Merge branch 'bug-empty-rewards-should-disappear' of https://github.com/ElimarCosta/canary into bug-empty-rewards-should-disappear

bateunatrave commented:
``` 
- -  #1  0x00007ffff7b0b50c in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
- -  No symbol table info available.
- -  #2  0x00005555555bd28e in DatabaseTasks::threadMain (this=0x555555b40280 <g_databaseTasks>) at -/src/databasetasks.cpp:45
- -          taskLockUnique = {_M_device = 0x555555b402b8 <g_databaseTasks+56>, _M_owns = true}
- -  #3  0x00007ffff7b10e6f in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
- -  No symbol table info available.
- -  #4  0x00007ffff78414a4 in start_thread (arg=0x7fffecf63700) at pthread_create.c:456
- -          __res = <optimized out>
- -          pd = 0x7fffecf63700
- -          now = <optimized out>
- -          unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140737168946944, 3302587052165691905, 140737187898382, 140737187898383, 140737160556544, 3, -3302549744502179327, -3302604840559314431}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
- -          not_first_call = <optimized out>
- -          pagesize_m1 = <optimized out>
- -          sp = <optimized out>
- -          freesize = <optimized out>
- -          __PRETTY_FUNCTION__ = "start_thread"
- -  #5  0x00007ffff4e1ad0f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
- -  No locals.
- -  
- -  Thread 3 (Thread 0x7fffed976700 (LWP 3613)):
- -  #0  pthread_cond_timedwait@@GLIBC_2.3.2 () at ../sysdeps/unix/sysv/linux/x86_64/pthread_cond_timedwait.S:225
- -  No locals.
- -  #1  0x00005555557ddc72 in __gthread_cond_timedwait (__abs_timeout=0x7fffed975e60, __mutex=<optimized out>, __cond=0x555555b40100 <g_scheduler+64>) at /usr/include/x86_64-linux-gnu/c++/6/bits/gthr-default.h:871
- -  No locals.
- -  #2  std::condition_variable::__wait_until_impl<std::chrono::duration<long, std::ratio<1l, 1000000000l> > > (__atime=..., __lock=..., this=0x555555b40100 <g_scheduler+64>) at /usr/include/c++/6/condition_variable:166
- -          __ts = {tv_sec = 1674129348, tv_nsec = 695215778}
- -  #3  std::condition_variable::wait_until<std::chrono::duration<long, std::ratio<1l, 1000000000l> > > (__atime=..., __lock=..., this=0x555555b40100 <g_scheduler+64>) at /usr/include/c++/6/condition_variable:106
- -  No locals.
- -  #4  Scheduler::threadMain (this=0x555555b400c0 <g_scheduler>) at -/src/scheduler.cpp:19
- -          ret = std::cv_status::no_timeout
- -          eventLockUnique = {_M_device = 0x555555b400d8 <g_scheduler+24>, _M_owns = true}
- -  #5  0x00007ffff7b10e6f in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
- -  No symbol table info available.
- -  #6  0x00007ffff78414a4 in start_thread (arg=0x7fffed976700) at pthread_create.c:456
- -          __res = <optimized out>
- -          pd = 0x7fffed976700
- -          now = <optimized out>
- -          unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140737179510528, 3302587052165691905, 140737488348126, 140737488348127, 140737171120128, 3, -3302547810156283391, -3302604840559314431}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
- -          not_first_call = <optimized out>
- -          pagesize_m1 = <optimized out>
- -          sp = <optimized out>
- -          freesize = <optimized out>
- -          __PRETTY_FUNCTION__ = "start_thread"
- -  #7  0x00007ffff4e1ad0f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
- -  No locals.
- -  
- -  Thread 2 (Thread 0x7fffee177700 (LWP 3612)):
- -  #0  0x000055555578a0f6 in Player::postAddNotification (this=0x7fffb70be730, thing=thing@entry=0x7fffb70be730, oldParent=oldParent@entry=0x7ffed661b1b0, index=index@entry=0, link=link@entry=LINK_NEAR) at -/src/player.cpp:3900
- -          container = 0x7fffc555f970
- -          __for_range = std::map with 4 elements = {[0 '\000'] = {container = 0x7fff8ba2a900, index = 0}, [1 '\001'] = {container = 0x7fff9a2db1b0, index = 0}, [2 '\002'] = {container = 0x7fff8fac7530, index = 0}, [4 '\004'] = {container = 0x7fffc555f970, index = 0}}
- -          containers = std::vector of length 0, capacity 0
- -          creature = <optimized out>
- -          item = <optimized out>
- -          requireListUpdate = <optimized out>
- -  #1  0x000055555580128d in Tile::postAddNotification (this=this@entry=0x7ffed6716ae0, thing=0x7fffb70be730, oldParent=oldParent@entry=0x7ffed661b1b0, index=index@entry=0, link=link@entry=LINK_OWNER) at -/src/tile.cpp:1384
- -          spectator = <optimized out>
- -          __for_range = std::unordered_set with 1 elements = {[0] = 0x7fffb70be730}
- -          spectators = std::unordered_set with 1 elements = {[0] = 0x7fffb70be730}
- -          creature = <optimized out>
- -          item = <optimized out>
- -  #2  0x000055555573e6a9 in Map::moveCreature (this=this@entry=0x555555b3f730 <g_game+80>, creature=..., newTile=..., forceTeleport=forceTeleport@entry=false) at -/src/map.cpp:305
- -          oldPos = {x = 33387, y = 32655, z = 3 '\003'}
- -          newPos = {x = 33387, y = 32656, z = 3 '\003'}
- -          teleport = <optimized out>
- -          spectators = std::unordered_set with 34 elements = {[0] = 0x7fff95d51c50, [1] = 0x7fff96207b90, [2] = 0x7fff98d678e0, [3] = 0x7fff95ebba50, [4] = 0x7fff98d81be0, [5] = 0x7fffba2d0210, [6] = 0x7fff96587690, [7] = 0x7fff9b3cbd60, [8] = 0x7fff9cc76640, [9] = 0x7fff8dc4ce40, [10] = 0x7fff9c80b2f0, [11] = 0x7fffb70be730, [12] = 0x7fffbc0cc180, [13] = 0x7fff900858e0, [14] = 0x7fffbc1e1c80, [15] = 0x7fff9c7597d0, [16] = 0x7fff9068bf50, [17] = 0x7fff9138c540, [18] = 0x7fffbbdfe1d0, [19] = 0x7fffbbc52280, [20] = 0x7fff90602c70, [21] = 0x7fff90087290, [22] = 0x7fff9cbc6480, [23] = 0x7fffb9df93a0, [24] = 0x7fffbbea67e0, [25] = 0x7fffbbc6df20, [26] = 0x7fff9c958660, [27] = 0x7fff956464f0, [28] = 0x7fff907286e0, [29] = 0x7fff97c43160, [30] = 0x7fffbbc68e10, [31] = 0x7fffbbbeaef0, [32] = 0x7fff90d3c290, [33] = 0x7fffbc1c5500}
- -          oldStackPosVector = std::vector of length 1, capacity 1 = {2}
- -          leaf = <optimized out>
- -          new_leaf = <optimized out>
- -          i = <optimized out>
- -  #3  0x00005555555f234a in Game::internalMoveCreature (this=0x555555b3f6e0 <g_game>, creature=..., toTile=..., flags=<optimized out>, flags@entry=32) at -/src/game.cpp:1471
- -          ret = <optimized out>
- -          index = 32767
- -          toItem = 0x7f037f90826b5823
- -          subCylinder = <optimized out>
- -          toCylinder = <optimized out>
- -          fromCylinder = <optimized out>
- -          n = <optimized out>
- -  #4  0x00005555555f2fda in Game::internalMoveCreature (this=<optimized out>, creature=creature@entry=0x7fffb70be730, direction=<optimized out>, flags=32) at -/src/game.cpp:1455
- -          destPos = {x = <optimized out>, y = 32656, z = <optimized out>}
- -          player = <optimized out>
- -          diagonalMovement = <optimized out>
- -          toTile = <optimized out>
- -  #5  0x0000555555857ce8 in Creature::onCreatureWalk (this=0x7fffb70be730) at -/src/creature.cpp:191
- -          ret = <optimized out>
- -          dir = DIRECTION_SOUTH
- -          flags = 32
- -  #6  0x00005555555c8544 in Game::checkCreatureWalk (this=0x555555b3f6e0 <g_game>, creatureId=<optimized out>) at -/src/game.cpp:5545
- -          creature = <optimized out>
- -  #7  0x0000555555857ba2 in Creature::addEventWalk (this=this@entry=0x7fffb70be730, firstStep=<optimized out>) at -/src/creature.cpp:283
- -          ticks = 1
- -  #8  0x0000555555857e39 in Creature::startAutoWalk (this=this@entry=0x7fffb70be730, listDir=std::forward_list = {...}) at -/src/creature.cpp:261
- -          size = <optimized out>
- -          this = 0x7fffb70be730
- -  #9  0x00005555555c86d7 in Game::playerMove (this=<optimized out>, playerId=<optimized out>, direction=<optimized out>) at -/src/game.cpp:2892
- -          player = 0x7fffb70be730
- -  #10 0x00005555557fc5da in std::function<void ()>::operator()() const (this=0x7fff9a13c120) at /usr/include/c++/6/functional:2127
- -  No locals.
- -  #11 Task::operator() (this=0x7fff9a13c110) at -/src/tasks.h:25
- -  No locals.
- -  #12 Dispatcher::threadMain (this=0x555555b401e0 <g_dispatcher>) at -/src/tasks.cpp:45
- -          task = 0x7fff9a13c110
- -          taskLockUnique = {_M_device = 0x555555b401f0 <g_dispatcher+16>, _M_owns = false}
- -  #13 0x00007ffff7b10e6f in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
- -  No symbol table info available.
- -  #14 0x00007ffff78414a4 in start_thread (arg=0x7fffee177700) at pthread_create.c:456
- -          __res = <optimized out>
- -          pd = 0x7fffee177700
- -          now = <optimized out>
- -          unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140737187903232, 3302587052165691905, 140737488348126, 140737488348127, 140737179512832, 3, -3302553307177551359, -3302604840559314431}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
- -          not_first_call = <optimized out>
- -          pagesize_m1 = <optimized out>
- -          sp = <optimized out>
- -          freesize = <optimized out>
- -          __PRETTY_FUNCTION__ = "start_thread"
- -  #15 0x00007ffff4e1ad0f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:97
- -  No locals.
- -  
- -  Thread 1 (Thread 0x7ffff7fd62c0 (LWP 3608)):
- -  #0  0x00007ffff4e1b303 in epoll_wait () at ../sysdeps/unix/syscall-template.S:84
- -  No locals.
- -  #1  0x00005555557e832c in boost::asio::detail::epoll_reactor::run (this=0x555555b81d60, block=block@entry=true, ops=...) at /usr/include/boost/asio/detail/impl/epoll_reactor.ipp:392
- -          timeout = <optimized out>
- -          events = {{events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}} <repeats 40 times>, {events = 0, data = {ptr = 0xf784ae2500000000, fd = 0, u32 = 0, u64 = 17835571898231029760}}, {events = 32767, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 4294967295, data = {ptr = 0x56a23890ffffffff, fd = -1, u32 = 4294967295, u64 = 6242614228910342143}}, {events = 21845, data = {ptr = 0x7fffffffdfc0, fd = -8256, u32 = 4294959040, u64 = 140737488347072}}, {events = 1439587528, data = {ptr = 0x5584e39100005555, fd = 21845, u32 = 21845, u64 = 6162300402059466069}}, {events = 21845, data = {ptr = 0xa1, fd = 161, u32 = 161, u64 = 161}}, {events = 208, data = {ptr = 0xffffdfa000000000, fd = 0, u32 = 0, u64 = 18446708477020602368}}, {events = 32767, data = {ptr = 0x100003200000045, fd = 69, u32 = 69, u64 = 72057808786292805}}, {events = 1438656118, data = {ptr = 0x200005555, fd = 21845, u32 = 21845, u64 = 8589956437}}, {events = 0, data = {ptr = 0x2, fd = 2, u32 = 2, u64 = 2}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 4294959008, data = {ptr = 0x100007fff, fd = 32767, u32 = 32767, u64 = 4295000063}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 0, data = {ptr = 0x7fffffffe1b0, fd = -7760, u32 = 4294959536, u64 = 140737488347568}}, {events = 1438680936, data = {ptr = 0x56a2389000005555, fd = 21845, u32 = 21845, u64 = 6242614224615396693}}, {events = 21845, data = {ptr = 0x555555c08768, fd = 1438680936, u32 = 1438680936, u64 = 93824999262056}}, {events = 1439587528, data = {ptr = 0x55b81d6000005555, fd = 21845, u32 = 21845, u64 = 6176719187092723029}}, {events = 21845, data = {ptr = 0x45, fd = 69, u32 = 69, u64 = 69}}, {events = 0, data = {ptr = 0x5584f81b00000000, fd = 0, u32 = 0, u64 = 6162322984997486592}}, {events = 21845, data = {ptr = 0x555555c02676, fd = 1438656118, u32 = 1438656118, u64 = 93824999237238}}, {events = 1432020427, data = {ptr = 0x5555, fd = 21845, u32 = 21845, u64 = 21845}}, {events = 1, data = {ptr = 0x7fffffffe1c0, fd = -7744, u32 = 4294959552, u64 = 140737488347584}}, {events = 4294959232, data = {ptr = 0x56a2389000007fff, fd = 32767, u32 = 32767, u64 = 6242614224615407615}}, {events = 21845, data = {ptr = 0x555556a23890, fd = 1453471888, u32 = 1453471888, u64 = 93825014053008}}, {events = 1432008583, data = {ptr = 0x55c0876800005555, fd = 21845, u32 = 21845, u64 = 6179087569498690901}}, {events = 21845, data = {ptr = 0x555555c02676, fd = 1438656118, u32 = 1438656118, u64 = 93824999237238}}, {events = 2, data = {ptr = 0x100000000, fd = 0, u32 = 0, u64 = 4294967296}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 1434769952, data = {ptr = 0x5555, fd = 21845, u32 = 21845, u64 = 21845}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 0, data = {ptr = 0x557ea7fb00000000, fd = 0, u32 = 0, u64 = 6160546036768047104}}, {events = 21845, data = {ptr = 0x3700000030, fd = 48, u32 = 48, u64 = 236223201328}}, {events = 1438656080, data = {ptr = 0x55c0266000005555, fd = 21845, u32 = 21845, u64 = 6178980882511058261}}, {events = 21845, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 1434769952, data = {ptr = 0xffffe1b000005555, fd = 21845, u32 = 21845, u64 = 18446710744763356501}}, {events = 32767, data = {ptr = 0x555555c08768, fd = 1438680936, u32 = 1438680936, u64 = 93824999262056}}, {events = 1434778483, data = {ptr = 0x5555, fd = 21845, u32 = 21845, u64 = 21845}}, {events = 0, data = {ptr = 0x7ffff6dbf0c8, fd = -153358136, u32 = 4141609160, u64 = 140737334997192}}, {events = 1438680936, data = {ptr = 0x55c0267600005555, fd = 21845, u32 = 21845, u64 = 6178980977000338773}}, {events = 21845, data = {ptr = 0x2, fd = 2, u32 = 2, u64 = 2}}, {events = 1, data = {ptr = 0x7fff, fd = 32767, u32 = 32767, u64 = 32767}}, {events = 0, data = {ptr = 0x55555584da20 <Connection::parseHeader(boost::system::error_code const&)>, fd = 1434769952, u32 = 1434769952, u64 = 93824995351072}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 1438656112, data = {ptr = 0x5584c51000005555, fd = 21845, u32 = 21845, u64 = 6162266862659851605}}, {events = 21845, data = {ptr = 0x555555c02660, fd = 1438656096, u32 = 1438656096, u64 = 93824999237216}}, {events = 4152655936, data = {ptr = 0x55c0870800007fff, fd = 32767, u32 = 32767, u64 = 6179087157181841407}}, {events = 21845, data = {ptr = 0x555555c02650, fd = 1438656080, u32 = 1438656080, u64 = 93824999237200}}, {events = 4294959552, data = {ptr = 0x55b81a7800007fff, fd = 32767, u32 = 32767, u64 = 6176715991637065727}}, {events = 21845, data = {ptr = 0x55555584cad7 <Connection::parsePacket(boost::system::error_code const&)+103>, fd = 1434766039, u32 = 1434766039, u64 = 93824995347159}}, {events = 1438656118, data = {ptr = 0x200005555, fd = 21845, u32 = 21845, u64 = 8589956437}}, {events = 0, data = {ptr = 0x55555584da20 <Connection::parseHeader(boost::system::error_code const&)>, fd = 1434769952, u32 = 1434769952, u64 = 93824995351072}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 3015236016, data = {ptr = 0xffffe31000007fff, fd = 32767, u32 = 32767, u64 = 18446712256591855615}}, {events = 32767, data = {ptr = 0x555555b81a20, fd = 1438128672, u32 = 1438128672, u64 = 93824998709792}}, {events = 12, data = {ptr = 0xffffe43800000000, fd = 0, u32 = 0, u64 = 18446713527902142464}}, {events = 32767, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 1438128760, data = {ptr = 0x5584f77600005555, fd = 21845, u32 = 21845, u64 = 6162322276327904597}}, {events = 21845, data = {ptr = 0x1, fd = 1, u32 = 1, u64 = 1}}, {events = 2, data = {ptr = 0x56a26f8000000000, fd = 0, u32 = 0, u64 = 6242674629035425792}}, {events = 21845, data = {ptr = 0x7fffffffe290, fd = -7536, u32 = 4294959760, u64 = 140737488347792}}, {events = 1438128672, data = {ptr = 0x55ce5ca000005555, fd = 21845, u32 = 21845, u64 = 6182981180690814293}}, {events = 21845, data = {ptr = 0x1, fd = 1, u32 = 1, u64 = 1}}, {events = 1439587528, data = {ptr = 0x5589f16000005555, fd = 21845, u32 = 21845, u64 = 6163722959652410709}}, {events = 21845, data = {ptr = 0x55555584fd05 <boost::asio::detail::reactive_socket_recv_op<boost::asio::mutable_buffers_1, boost::asio::detail::read_op<boost::asio::basic_stream_socket<boost::asio::ip::tcp, boost::asio::stream_socket_service<boost::asio::ip::tcp> >, boost::asio::mutable_buffers_1, boost::asio::detail::transfer_all_t, std::_Bind<std::_Mem_fn<void (Connection::*)(boost::system::error_code const&)> (std::shared_ptr<Connection>, std::_Placeholder<1>)> > >::do_complete(boost::asio::detail::task_io_service*, boost::asio::detail::task_io_service_operation*, boost::system::error_code const&, unsigned long)+245>, fd = 1434778885, u32 = 1434778885, u64 = 93824995360005}}, {events = 4294959760, data = {ptr = 0x7fff, fd = 32767, u32 = 32767, u64 = 32767}}, {events = 0, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 1451317504, data = {ptr = 0x100005555, fd = 21845, u32 = 21845, u64 = 4294989141}}, {events = 21845, data = {ptr = 0x555555c02676, fd = 1438656118, u32 = 1438656118, u64 = 93824999237238}}, {events = 2, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 0, data = {ptr = 0x2, fd = 2, u32 = 2, u64 = 2}}, {events = 4294959888, data = {ptr = 0x55b81a2000007fff, fd = 32767, u32 = 32767, u64 = 6176715613679943679}}, {events = 21845, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 4294960184, data = {ptr = 0x5584c52100007fff, fd = 32767, u32 = 32767, u64 = 6162266935674306559}}, {events = 21845, data = {ptr = 0x555555b81a78, fd = 1438128760, u32 = 1438128760, u64 = 93824998709880}}, {events = 1434778885, data = {ptr = 0xffffe31000005555, fd = 21845, u32 = 21845, u64 = 18446712256591844693}}, {events = 32767, data = {ptr = 0x7fffffffe340, fd = -7360, u32 = 4294959936, u64 = 140737488347968}}, {events = 1438128672, data = {ptr = 0x5584f15d00005555, fd = 21845, u32 = 21845, u64 = 6162315571883955541}}, {events = 21845, data = {ptr = 0x555555c02660, fd = 1438656096, u32 = 1438656096, u64 = 93824999237216}}, {events = 1438656080, data = {ptr = 0xffffe34000005555, fd = 21845, u32 = 21845, u64 = 18446712462750274901}}, {events = 32767, data = {ptr = 0x0, fd = 0, u32 = 0, u64 = 0}}, {events = 0, data = {ptr = 0x5584ca7000000000, fd = 0, u32 = 0, u64 = 6162272772534829056}}, {events = 21845, data = {ptr = 0x55555584c510 <Connection::handleTimeout(std::weak_ptr<Connection>, boost::system::error_code const&)>, fd = 1434764560, u32 = 1434764560, u64 = 93824995345680}}, {events = 1438656096, data = {ptr = 0x55c0265000005555, fd = 21845, u32 = 21845, u64 = 6178980813791581525}}, {events = 21845, data = {ptr = 0x55550000007d, fd = 125, u32 = 125, u64 = 93823560581245}}}
- -          num_events = <optimized out>
- -          check_timers = <optimized out>
- -  #2  0x00005555557e51b4 in boost::asio::detail::task_io_service::do_run_one (ec=..., this_thread=..., lock=..., this=0x555555b81a20) at /usr/include/boost/asio/detail/impl/task_io_service.ipp:356
- -          more_handlers = <optimized out>
- -  #3  boost::asio::detail::task_io_service::run (ec=..., this=0x555555b81a20) at /usr/include/boost/asio/detail/impl/task_io_service.ipp:149
- -          this_thread = {<boost::asio::detail::thread_info_base> = {<boost::asio::detail::noncopyable> = {<No data fields>}, reusable_memory_ = 0x555556a23040}, private_op_queue = {<boost::asio::detail::noncopyable> = {<No data fields>}, front_ = 0x0, back_ = 0x0}, private_outstanding_work = 0}
- -          ctx = {<boost::asio::detail::noncopyable> = {<No data fields>}, key_ = 0x555555b81a20, value_ = 0x7fffffffe430, next_ = 0x0}
- -          lock = {<boost::asio::detail::noncopyable> = {<No data fields>}, mutex_ = @0x555555b81a50, locked_ = false}
- -          n = <optimized out>
- -  #4  boost::asio::io_service::run (this=<optimized out>) at /usr/include/boost/asio/impl/io_service.ipp:59
- -          ec = {m_val = 0, m_cat = 0x7ffff6dbf0c8}
- -  #5  ServiceManager::run (this=<optimized out>) at -/src/server.cpp:30
- -  No locals.
- -  #6  0x00005555555922cb in main (argc=<optimized out>, argv=<optimized out>) at -/src/otserv.cpp:92
- -          serviceManager = {acceptors = std::unordered_map with 2 elements = {[7171] = std::shared_ptr (count 5, weak 1) 0x7fffa67aeb40, [7172] = std::shared_ptr (count 157, weak 1) 0x7fffb4c37170}, io_service = {<boost::asio::detail::noncopyable> = {<No data fields>}, service_registry_ = 0x555555b819d0, impl_ = @0x555555b81a20}, signals = {set = {<boost::asio::basic_io_object<boost::asio::signal_set_service, false>> = {service = @0x555555b81b00, implementation = {queue_ = {<boost::asio::detail::noncopyable> = {<No data fields>}, front_ = 0x555555b82000, back_ = 0x555555b82000}, signals_ = 0x555555b81fc0}}, <No data fields>}}, death_timer = {<boost::asio::basic_io_object<boost::asio::deadline_timer_service<boost::posix_time::ptime, boost::asio::time_traits<boost::posix_time::ptime> >, false>> = {service = @0x555555b82050, implementation = {<boost::asio::detail::noncopyable> = {<No data fields>}, expiry = {<boost::date_time::base_time<boost::posix_time::ptime, boost::date_time::counted_time_system<boost::date_time::counted_time_rep<boost::posix_time::millisec_posix_time_system_config> > >> = {<boost::operators_impl::less_than_comparable<boost::posix_time::ptime, boost::operators_impl::equality_comparable<boost::posix_time::ptime, boost::posix_time::ptime, boost::operators_impl::operators_detail::empty_base<boost::posix_time::ptime>, boost::operators_impl::operators_detail::false_t>, boost::operators_impl::operators_detail::empty_base<boost::posix_time::ptime>, boost::operators_impl::operators_detail::true_t>> = {<boost::operators_impl::less_than_comparable1<boost::posix_time::ptime, boost::operators_impl::equality_comparable<boost::posix_time::ptime, boost::posix_time::ptime, boost::operators_impl::operators_detail::empty_base<boost::posix_time::ptime>, boost::operators_impl::operators_detail::false_t> >> = {<boost::operators_impl::equality_comparable<boost::posix_time::ptime, boost::posix_time::ptime, boost::operators_impl::operators_detail::empty_base<boost::posix_time::ptime>, boost::operators_impl::operators_detail::false_t>> = {<boost::operators_impl::equality_comparable1<boost::posix_time::ptime, boost::operators_impl::operators_detail::empty_base<boost::posix_time::ptime> >> = {<boost::operators_impl::operators_detail::empty_base<boost::posix_time::ptime>> = {<No data fields>}, <No data fields>}, <No data fields>}, <No data fields>}, <No data fields>}, time_ = {time_count_ = {value_ = 9223372036854775806}}}, <No data fields>}, might_have_pending_waits = false, timer_data = {op_queue_ = {<boost::asio::detail::noncopyable> = {<No data fields>}, front_ = 0x0, back_ = 0x0}, heap_index_ = 1, next_ = 0x0, prev_ = 0x0}}}, <No data fields>}, running = true}
 
`
``` 

o pull ta funcionando mais apos certo tempo comecei a ter esse crash mais não entendi ainda como ele ocorre no lado do bau de recompensa

dudantas reviewed (changes requested):
Resolver os problemas do [sonar](https://sonarcloud.io/project/issues?id=opentibiabr_canary&pullRequest=779&resolved=false&types=BUG), podem causar graves inconsistências:
![image](https://user-images.githubusercontent.com/8551443/213499658-5d342fc5-1822-4e75-92fd-b8858daa8ebe.png)

dudantas committed:
Merge branch 'main' into bug-empty-rewards-should-disappear

ElimarCosta commented:
Sim, essa pr ta com alguns problemas ainda, tem varios bugs relacionados e eu to sem tempo essa semana por isso marquei como draft.

ElimarCosta committed:
fix reward bag limit

ElimarCosta committed:
fix reward bag limit

ElimarCosta committed:
Merge branch 'main' into bug-empty-rewards-should-disappear

ElimarCosta committed:
fix reward bag

ElimarCosta committed:
fix reward bag

bateunatrave commented:
/iologindata.cpp: In static member function ‘static void IOLoginData::insertItensIntoRewardBag(IOLoginData::ItemMap&)’:
iologindata.cpp:745:32: error: ‘std::views’ has not been declared
     for (const auto& it : std::views::reverse(itemMap)) {

ElimarCosta committed:
Merge branch 'main' into bug-empty-rewards-should-disappear

dudantas commented:
> /iologindata.cpp: In static member function ‘static void IOLoginData::insertItensIntoRewardBag(IOLoginData::ItemMap&)’: iologindata.cpp:745:32: error: ‘std::views’ has not been declared for (const auto& it : std::views::reverse(itemMap)) {

You need to use a compiler that supports c++20 or more, it seems to me that yours doesn't support it, what version of the compiler are you using? visual studio needs to be 17.2 or higher and gcc 11 or higher

bateunatrave commented:
> > /iologindata.cpp: In static member function ‘static void IOLoginData::insertItensIntoRewardBag(IOLoginData::ItemMap&)’: iologindata.cpp:745:32: error: ‘std::views’ has not been declared for (const auto& it : std::views::reverse(itemMap)) {
> 
> You need to use a compiler that supports c++20 or more, it seems to me that yours doesn't support it, what version of the compiler are you using? visual studio needs to be 17.2 or higher and gcc 11 or higher

I'm compiling for linux gcc 12

dudantas commented:
> > > /iologindata.cpp: In static member function ‘static void IOLoginData::insertItensIntoRewardBag(IOLoginData::ItemMap&)’: iologindata.cpp:745:32: error: ‘std::views’ has not been declared for (const auto& it : std::views::reverse(itemMap)) {
> > 
> > 
> > You need to use a compiler that supports c++20 or more, it seems to me that yours doesn't support it, what version of the compiler are you using? visual studio needs to be 17.2 or higher and gcc 11 or higher
> 
> I'm compiling for linux

This is not enough. What compiler version are you using? Is it gcc? If so, it needs to be version 11 or higher.

dudantas commented on src/io/iologindata.cpp:
```diff
@@ -738,6 +692,58 @@ bool IOLoginData::loadPlayer(Player* player, DBResult_ptr result)
   return true;
 }
 
+void IOLoginData::loadRewardItems(Player *player, DBResult_ptr &result) {
```
```suggestion
void IOLoginData::loadRewardItems(Player *player, DBResult_ptr &result) {
```
You can migrate this new functions to here: https://github.com/opentibiabr/canary/blob/97f3e8c7c5c8023d48df7b13e9fa531e318966d5/src/io/functions/iologindata_load_player.cpp#L28

dudantas committed:
Some indent

dudantas committed:
Some indent

dudantas committed:
Merge remote-tracking branch 'upstream/main' into pr/779

dudantas committed:
Merge remote-tracking branch 'upstream/main' into pr/779

ElimarCosta committed:
mover para load player

ElimarCosta committed:
mover para load player

ElimarCosta committed:
move to save

ElimarCosta committed:
move to save

ElimarCosta committed:
ajustes

ElimarCosta committed:
ajustes

Commit:
fix: Canary NPC example addModule missing parameters (#842)

Now the NPCs are displayed correctly and everything appears to be functional.
Previously, they were not working in Ghost mode and this is what they looked like.

If you have custom NPCs like I do, and created them some time ago, you may need to update them to function correctly.
Change "npcHandler:addModule(FocusModule:new())" to
"npcHandler:addModule(FocusModule:new(), npcConfig.name, true, true, true)"

dudantas committed:
improve: rework of Item Attributes for better stability/maintability (#827)

This brings in a major overhaul of the Item Attributes component, addressing long-standing issues in the TFS and OpenTibiaServer project. Adopting class inheritance and harnessing the latest capabilities of C++, this update resolves specific crashes and streamlines future changes to the code. The Attributes class has been separated into its own file, making the code more readable and easier to understand, while also removing any confusing and redundant functions from the Item class. This results in a more stable, maintainable, and user-friendly solution for Item Attributes.

Simplified code by moving to shared_ptr for memory management, removing redundant functions, and revamping key classes such as Attributes, ItemAttribute, and CustomAttribute. The deprecated "set/getSpecialAttribute" functions have been replaced with more efficient "set/getCustomAttribute." This simplified approach leads to better code maintenance and increased stability, for example, Hirelings now utilize the CustomAttribute.

dudantas committed:
fix: client debug and show not targeting player when dying near temple (#841)

The issue arose because the multi-floor was not properly verified, causing incorrect identification of some creatures.

dudantas committed:
fix: some errors and nil value in lua scripts (#830)

murilo09 committed:
feat: added outfit and mount preview feature to the store (#773)

This change adds the new outfit and mount preview feature in the Tibia game store. In addition to the new interface, which offers a better structured and layout, players can now try out all the available outfits and mounts before purchasing them.

ElimarCosta committed:
fix: using a shovel on a stonepile should move to downstairs (#757)

murilo09 committed:
fix: NPC Ongulf (#759)

In the https://github.com/opentibiabr/otservbr-global/pull/781 fixing the `The New Frontier Quest`, mistakenly replaced Ongulf's script with Leeland's script. Reverting to the correct script.

murilo09 committed:
fix: remove money lua function should remove money from bank too (#760)

ElimarCosta committed:
fix: players shouldn't pathfind through fields (#756)

ElimarCosta committed:
ajustes

ElimarCosta committed:
ajustes

ElimarCosta committed:
update banco

ElimarCosta committed:
fix merge

ElimarCosta committed:
Merge branch 'bug-empty-rewards-should-disappear' of https://github.com/ElimarCosta/canary into bug-empty-rewards-should-disappear

ElimarCosta committed:
ajuste sonar

ElimarCosta committed:
teste

ElimarCosta committed:
teste

ElimarCosta committed:
fix

ElimarCosta commented:
moved to #846
